runelite-client/src/main/java/net/runelite/client/plugins/questlist/QuestListPlugin.java

```diff
@@ -0,0 +1,401 @@
+
+	private ChatboxTextInput searchInput = null;
+	private Widget questSearchButton;
+	private Widget questHideCompletedButton;
+
+	private EnumMap<QuestContainer, Collection<QuestWidget>> questSet;
+
+	private boolean hideCompleted = false;
+
+	@Data
+	private static class QuestWidget
+	{
+		private String title;
+		private Widget quest;
+	}
+
+	@AllArgsConstructor
+	@Getter
+	private enum QuestContainer
+	{
+		FREE_QUESTS(WidgetInfo.QUESTLIST_FREE_CONTAINER),
+		MEMBER_QUESTS(WidgetInfo.QUESTLIST_MEMBERS_CONTAINER),
+		MINI_QUESTS(WidgetInfo.QUESTLIST_MINIQUEST_CONTAINER);
+
+		private final WidgetInfo widgetInfo;
+	}
+
+	private enum QuestState
+	{
+		NOT_STARTED(0xff0000), IN_PROGRESS(0xffff00), COMPLETE(0xdc10d);
+
+		@Getter
+		private final int color;
+
+		QuestState(int color)
+		{
+			this.color = color;
+		}
+
+		public static QuestState getByColor(int color)
+		{
+			for (QuestState value : values())
+			{
+				if (value.getColor() == color)
+				{
+					return value;
+				}
+			}
+
+			return null;
+		}
+	}
+
+	@Provides
+	QuestListConfig getConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(QuestListConfig.class);
+	}
+
+	@Subscribe
+	public void onGameStateChanged(GameStateChanged e)
+	{
+		if (e.getGameState() == GameState.LOGGING_IN)
+		{
+			hideCompleted = false;
+		}
+	}
+
+	@Subscribe
+	public void onWidgetLoaded(WidgetLoaded widgetLoaded)
+	{
+		if (widgetLoaded.getGroupId() == WidgetID.QUESTLIST_GROUP_ID)
+		{
+			Widget header = client.getWidget(WidgetInfo.QUESTLIST_BOX);
+			if (header != null)
+			{
+				questSearchButton = header.createChild(-1, WidgetType.GRAPHIC);
+				questSearchButton.setSpriteId(SpriteID.GE_SEARCH);
+				questSearchButton.setOriginalWidth(18);
+				questSearchButton.setOriginalHeight(17);
+				questSearchButton.setOriginalX(165);
+				questSearchButton.setOriginalY(0);
+				questSearchButton.setHasListener(true);
+				questSearchButton.setAction(1, MENU_OPEN);
+				questSearchButton.setOnOpListener((JavaScriptCallback) this::menuOpen);
+				questSearchButton.setName(MENU_SEARCH);
+				questSearchButton.revalidate();
+
+				questHideCompletedButton = header.createChild(-1, WidgetType.GRAPHIC);
+				redrawHideCompletedButton();
+
+				questHideCompletedButton.setOriginalWidth(18);
+				questHideCompletedButton.setOriginalHeight(17);
+				questHideCompletedButton.setOriginalX(145);
+				questHideCompletedButton.setOriginalY(0);
+				questHideCompletedButton.setHasListener(true);
+				questHideCompletedButton.setOnOpListener((JavaScriptCallback) this::menuOpen);
+				questHideCompletedButton.setName(MENU_COMPLETED);
+
+				questHideCompletedButton.revalidate();
+
+				questSet = new EnumMap<>(QuestContainer.class);
+
+				if (!header.isHidden())
+				{
+					updateFilter();
+				}
+			}
+		}
+	}
+
+	@Subscribe
+	public void onVarbitChanged(VarbitChanged varbitChanged)
+	{
+		if (isChatboxOpen() && !isOnQuestTab())
+		{
+			chatboxPanelManager.close();
```

Comment:
this should not close inputs randomly

Reply:
This is not randomly. I am closing the search dialog when the user navigates away from the quest tab, either to e.g. the achievement tab or the inventory tab. At that point the search dialog is no longer relevant. It is closed only if the currently open search dialog is the one this plugin created (see `isChatboxOpen`). Is this not desired behavior?

Reply:
ah, I misread it as isChatboxOpen being's chatboxPanelManagers